### PR TITLE
Misc.

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

**Comment by fisx** [comment]
ready for review!

File: src/Persistent/Implementation/AcidState.hs [hunk 1]
```diff
@@ -29,23 +31,39 @@ import Persistent.Api
...
+
+    cls st tid = do
+        killThread tid
+        createCheckpointAndClose st
 
 mkRunPersistInMemory :: IO RunPersist
 mkRunPersistInMemory =
     mkRunPersistGeneric "acid-state (memory)"
-        openMemoryState closeAcidState
+        (\aulaData -> (, ()) <$> openMemoryState aulaData)
```
Comment (andorp) [inline src/Persistent/Implementation/AcidState.hs]: `fmap` ((,) ()) . openMemoryState` ?

File: src/Persistent/Implementation/AcidState.hs [hunk 2]
```diff
@@ -29,23 +31,39 @@ import Persistent.Api
...
 
 mkRunPersistOnDisk :: Config -> IO RunPersist
 mkRunPersistOnDisk cfg =
-    mkRunPersistGeneric "acid-state (disk)"
-        (openLocalStateFrom $ cfg ^. dbPath) createCheckpointAndClose
+    mkRunPersistGeneric "acid-state (disk)" opn cls
+  where
+    opn aulaData = do
+        st <- openLocalStateFrom (cfg ^. persistConfig . dbPath) aulaData
+        tid <- forkIO . forever $ do
```
Comment (andorp) [inline src/Persistent/Implementation/AcidState.hs]: Please add exception handling, and restart the thread if something terrible happened. Also some log entry would be fine here, about the exception.

**Comment by fisx** [comment]
like this?

**Comment by andorp** [comment]
It looks good, but some imports are missing, do you want me to fix that?

**Pull request merged**